src/shape.rs

```diff
@@ -1194,9 +1204,15 @@ impl ShapeLine {
                             }
                             word_range_width += word_width;
                             continue;
-                        } else if wrap == Wrap::Glyph
+                        }
+
+                        if wrap == Wrap::Glyph
                             // Make sure that the word is able to fit on it's own line, if not, fall back to Glyph wrapping.
                             || (wrap == Wrap::WordOrGlyph && word_width > line_width)
+                            // If we're on the first line
+                            || (wrap == Wrap::WordOrGlyph && visual_lines.len() == 0 &&
+                            // and we can't fit the rest of this word on the line
+                            (current_visual_line.w + word_width > line_width))
```

Comment:
For my own understanding - in this case of the first line, should we be adjusting `line_width` by `first_line_indent`? Saw that `current_line_width` accounts for this but `line_width` is more general?

Reply:
`current_visual_line.w` already accounts for the indent width, so this shouldn't need further adjustment.